studygroups/views/api.py

```diff
@@ -911,8 +920,9 @@ def get(self, request):
 class FinalReportListView(View):
     def get(self, request):
         today = timezone.now().replace(hour=0, minute=0, second=0, microsecond=0)
-        studygroups = StudyGroup.objects.published().filter(end_date__lt=today)
-        studygroups = filter_studygroups_with_survey_responses(studygroups)
+        completed_studygroups = StudyGroup.objects.published().filter(end_date__lt=today)
+        with_responses = filter(lambda sg: sg.learnersurveyresponse_set.count() > 0, completed_studygroups)
```

Comment:
@s-kennedy this will result in N+1 db queries for N completed studygroups. Where are we using this view? It might be okay, but an annotation would be better in this case:

```python
StudyGroup.objects.annotate(surveys=Count('learnersurveyresponse')).filter(surveys__gt=0)
```

Reply:
This is used in for the "Recent Successes" block in the dashboard, its' just a modification of the `filter_studygroups_with_survey_responses` function that was there before, except that function was sorting by number of responses, and we want to sort by date so we always get the latest ones. The function is also used for the community digest but I guess we don't care as much about performance there since it's generated in a background job.